libs/common/src/schemas/portal-user.update.test.ts

```diff
@@ -0,0 +1,57 @@
+import { ObjectId } from 'mongodb';
+import { generatePortalUserAuditDatabaseRecord } from '../change-stream';
+import { UPDATE } from './portal-user';
+import { withSchemaTests } from '../test-helpers';
+
+describe('PORTAL_USER', () => {
+  describe('UPDATE', () => {
+    withSchemaTests({
+      successTestCases: getSuccessTestCases(),
+      failureTestCases: getFailureTestCases(),
+      schema: UPDATE,
+    });
+  });
+});
+
+function getSuccessTestCases() {
+  return [
+    { description: 'a valid user', aTestCase: () => aValidPortalUser() },
+    {
+      description: 'a partial user',
+      aTestCase: () => {
+        const { username: _username, ...rest } = aValidPortalUser();
+        return rest;
+      },
+    },
+  ];
+}
+
+function getFailureTestCases() {
+  return [
+    { description: 'a string', aTestCase: () => 'string' },
+    { description: 'an object', aTestCase: () => ({ An: 'object' }) },
+    { description: 'an array', aTestCase: () => ['array'] },
+    { description: 'a user with an additional parameter', aTestCase: () => ({ ...aValidPortalUser(), invalidField: true }) },
+  ];
+}
+
+function aValidPortalUser() {
```

Comment:
Do you think it's worth moving this into a shared mock/fixture, since it's repeated twice 

Maybe a bigger piece as I imagine a lot of portal user objects are duplicated without the `aValidPortalUser` wrapper

Reply:
It's an incredibly large piece of work that would be prone to error (I've had this thought a lot) as the portal user type is incredibly transient (there are several). The first stage of this work is actually typing portal users properly in our codebase so we can work out what's what (which is an ongoing process)!